src/avatar.component.ts

```diff
@@ -135,8 +135,8 @@ export class AvatarComponent implements OnChanges {
       this.avatarStyle = this._initialsStyle(avatarSource.sourceId);
     } else {
       this.avatarStyle = this._imageStyle();
-      if (utils.isAsyncAvatar(avatarSource.sourceType)) {
-        this._fetchAsyncAvatar(avatarSource);
+      if (avatarSource.isAsync) {
```

Comment:
If we use an abstract class to define a source as asynchronous, the check would look like this : 
```ts
if(avatarSource instanceof AsyncSource){
    this._fetchAsyncAvatar(<AsyncSource>avatarSource);
} else {
this.src = avatarSource.getAvatar(this.size);
}
```